spec/unit/puppet/provider/powershell_spec.rb

```diff
@@ -0,0 +1,157 @@
+#! /usr/bin/env ruby
+require 'spec_helper'
+require 'puppet/type'
+require 'puppet/type/base_dsc'
+require 'fileutils'
+
+describe Puppet::Type.type(:base_dsc).provider(:powershell) do
+
+  # Override initialize so we don't call the symlink set
+  class Puppet::Type::Base_dsc::ProviderPowershell
+    def initialize(value={})
+      super(value)
+    end
+  end
+
+  let (:resource) { Puppet::Type.type(:base_dsc).new(:provider => :powershell, :name => "windows_provider") }
+  let (:provider) { resource.provider }
+
+  before :each do
+    resource.provider = provider
```

Comment:
I don't really understand what this is for?

In my proposed changes I've axed it, since you can just use `subject` everywhere (and it's a different object instance within each test which I *think* is what you were looking for?).